### PR TITLE
Update README with Pinning information

### LINKED ISSUE #18: Certificate pinning support
A common way to add ssl pinning support in iOS apps is to use the `NSURLConnectionDelegate` to define a custom `didReceiveAuthenticationChallenge`. How to add this security feature to Siesta framework?

For a complete example see the [OWASP website](https://www.owasp.org/index.php/Certificate_and_Public_Key_Pinning#iOS).

### PR DESCRIPTION
**Pull request by lifeisfoo**
fix #18

### CODE REVIEW

**Comment by pcantrell** [comment]
Thanks for this, Alessandro!

Two things before I merge:

- This is too specific for the REAME. Let's move it to the [user guide](https://github.com/bustoutsolutions/siesta/tree/master/Docs). Create a new “Security” page for it, because we may want to add other sections as well (e.g. authentication, pitfalls, etc.).
- I'm noting a few minor copy edits in the text.

File: README.md [hunk 1]
```diff
@@ -289,6 +290,34 @@ The same functionality. Yes, really.
...
+```
+let certificatePinningSession = NSURLSession(
+    configuration: NSURLSessionConfiguration.ephemeralSessionConfiguration(),
+    delegate: MyCustomSessionPinningDelegate(),
+    delegateQueue: nil)
+let myService = Service(baseURL: "http://what.ever", networking: certificatePinningSession)
+```
+
+For an example code see the [OWASP guide](https://www.owasp.org/index.php/Certificate_and_Public_Key_Pinning#iOS
+) (it's for NSURLConnection but code is very similar).
```
Comment (pcantrell) [inline README.md]: "(it’s for NSURLConnection" should be a new sentence.

File: README.md [hunk 2]
```diff
@@ -289,6 +290,34 @@ The same functionality. Yes, really.
...
+
+### 2. Alamofire
+
+If you are using Siesta with Alamofire as a [networking provider](http://bustoutsolutions.github.io/siesta/api/Protocols/NetworkingProvider.html), you could use the [Alamofire security utils](https://github.com/Alamofire/Alamofire#security).
+
+### 3. TrustKit
+
+[TrustKit](https://github.com/datatheorem/TrustKit) "_will by default swizzle the App's NSURLSession and NSURLConnection delegates to verify the server's certificate against the configured pinning policy, whenever an HTTPS connection is initiated_".
+Using it you could have certificates pinning "_without even modifying the App's source code_".
```
Comment (pcantrell) [inline README.md]: No need for both quotes and italics, plus tightening up the text a little:

   It provides certificate pinning without modifying the App's source code.

**Comment by lifeisfoo** [comment]
Ok, but I think that a short note in the README (with a link to the guide) could be also useful.
Every time I look for something, I start from the README :).

**Comment by pcantrell** [comment]
My apologies — I should have explained that the `gh-pages` branch is generated from `master`; all you need to do is add the file to the `Docs` directory on `master`. (You don't need to add the jekyll header or the next page link. Those are also generated.)

I hate to make you keep jumping through hoops, so I can take care of that. I’ll just merge this PR and move it to the user guide myself.  Thanks again for writing this up!

**Comment by lifeisfoo** [comment]
No problem, but add this information in the Contributing section of the README ;).

**Pull request merged**